docs/src/tutorials/manifold.md

```diff
@@ -0,0 +1,198 @@
+
+Let's look at this for an example. The exponential map $\exp_p\colon T_p\mathcal M \to \mathcal M$ that maps a tangent vector $X\in T_p\mathcal M$ from the tangent space at $p\in \mathcal M$ to the manifold.
+The function [`exp`](@ref exp(M::Manifold, p, X)) has to know the manifold `M`, the point `p` and the tangent vector `X` as input, so to compute the resulting point `q` would be calling
+
+```julia
+q = exp(M, p, X)
+```
+
+If you already have allocated memory to the variable that should store the result, it is better to perform the computations directly in that memory and avoid reallocations. For example
+
+```julia
+q = similar(p)
+exp!(M, q, p, X)
+```
+
+calls [`exp!`](@ref exp!(M::Manifold, q, p, X)), which modifies its input `q` and returns the resulting point in there.
+Actually these two lines are the default implementation for [`exp`](@ref exp(M::Manifold, p, X)).
+Note that for a unified interface, the manifold `M` is _always_ the first parameter, and the variable the result will be stored to in the mutating variants is _always_ the second parameter.
+
+Long story short: if possible, implement the mutating version [`exp!`](@ref exp!(M::Manifold, q, p, X)), you get the [`exp`](@ref exp(M::Manifold, p, X)) for free.
+Many functions that build upon basic functions employ the mutating variant, too, to avoid reallocations.
+
+## [Startup](@id manifold-tutorial-startup)
+
+As a start, let's load `ManifoldsBase.jl` and import the functions we consider throughout this tutorial.
+For implementing a manifold, loading the interface should suffice for quite some time.
+
+```@example manifold-tutorial
+using ManifoldsBase, LinearAlgebra, Test
+import ManifoldsBase: check_manifold_point, check_tangent_vector, manifold_dimension, exp!
+```
+
+## [Goal](@id manifold-tutorial-task)
+
+As an example, let's implement the sphere, but with a radius $r$. Since this radius is a property inherent to the manifold.
+The second information, we want to store is the dimension of the sphere, for example whether it's the 1-sphere, i.e. the circle, represented by vectors $p\in\mathbb R^2$ or the 2-sphere in $\mathbb R^3$.
+Since the latter might be something we want to [dispatch](https://en.wikipedia.org/wiki/Multiple_dispatch) on, we model it as a parameter of the type. Hence we define
+
+```@example manifold-tutorial
+"""
+    MySphere{N} <: Manifold{ℝ}
+
+Define an `n`-sphere of radius `r`. Construct by `MySphere(radius,n)
+"""
+struct MySphere{N} <: Manifold{ManifoldsBase.ℝ} where {N}
+    radius::Float64
+end
+MySphere(radius, n) = MySphere{n}(radius)
+Base.show(io::IO, M::MySphere{n}) where {n} = print(io, "MySphere($(M.radius),$n)")
+nothing #hide
+```
+
+Here, the last line just provides a nicer print of a variable of that type
+The manifold should only contain information that it might require independently of its points or tangents.
```

Comment:
Might be worth adding that it should always be possible to determine the dimension of a manifold based on the manifold object only.

Reply:
Maybe I phrase that even further, when introducing the struct, which thing the struct should be able to do.